server/routes/render.js

```diff
@@ -73,26 +65,77 @@ const render = (request, response) => {
   const component = cache[pathToSource];
 
   // TODO test this behaviour
+  const markup = component.render(props, toStaticMarkup);
+  const componentKey = component.element.key;
+  return{
+    markup,
+    slug: componentKey,
+    files: {
+      [component.element.key]: getHashedFilename(componentKey),
+      vendor: getHashedFilename('vendor'),
+      stylesheet: getHashedFilename(componentKey, 'css')
+    }
+  };
+}
+
+const render = (request, response) => {
+  let defaultProps = JSON.stringify({
+    _serverContext: {
+      location: ''
+    }
+  });
+
+  let {
+    toStaticMarkup = false,
+    path: pathToSource,
+    serializedProps = defaultProps
+  } = request.body;
+
+  let props = JSON.parse(serializedProps);  
+
+  // TODO test this behaviour
   try {
-    const markup = component.render(props, toStaticMarkup);
-    const componentKey = component.element.key;
-    response.send({
-      markup,
-      slug: componentKey,
-      files: {
-        [component.element.key]: getHashedFilename(componentKey),
-        vendor: getHashedFilename('vendor'),
-        stylesheet: getHashedFilename(componentKey, 'css')
-      }
-    });
+    response.send(renderComponent(pathToSource, props, toStaticMarkup));
   } catch(e) {
     rollbar.handleError(e, request);
     return response.status(400).send({ 
       error: `Error rendering component: '${pathToSource}'`, 
       stack: e.stack 
     });
   }
-}
+} 
 
+const renderPage = (request, response) => {
+  try {
+    let component, initialState, context = {};
+
+    const validRoute = routes.some(route => {
+      const match = matchPath(request.url, route);
+      if (match) {
+        let props = {_serverContext: {location: request.url}, form_options: {}, options: {serverRender: true}};
+        initialState = Object.assign({}, props);
+        component = renderComponent(route.widgetPath, props, false);
+      }
+      return match
+    })
+
+    if (!validRoute) {
+      return response.status(404).send('Page not found');
+    }
+
+    response.send('<!doctype html>' + ReactDOMServer.renderToString(
```

Comment:
I did not know this was a thing... https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/HTML5/Introduction_to_HTML5#Declaring_that_the_document_contains_HTML5_mark-up_with_the_HTML5_doctype